### PR TITLE
Rel to #10927 - Looker DataModel

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes:

This PR is related to the EPIC #10927

This is the major chunk of work to ingest data models from Looker. Note that the git file parsing needs to be validated with the helps of some community users, but the main ingestion of dashboards and explores happens properly

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Review by nahuelverdugo — APPROVED** [review]
LGTM 🚀

File: ingestion/src/metadata/ingestion/source/dashboard/looker/columns.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+def get_column_from_model(model: Union[LookmlModelExplore, LookMlView]) -> List[Column]:
+    """
+    Obtain the column (measures and dimensions) from the models
+    """
+    columns = []
+    all_fields = (model.fields.dimensions or []) + (model.fields.measures or [])
+    for field in cast(Sequence[LookmlModelExploreField], all_fields):
+        columns.append(
+            Column(
```
Comment (nahuelverdugo) [inline ingestion/src/metadata/ingestion/source/dashboard/looker/columns.py:98]: Maybe we can try with this `displayName`: https://github.com/looker-open-source/sdk-codegen/blob/main/python/looker_sdk/sdk/api40/models.py#L6267

**Review by TeddyCr — COMMENTED** [review]
@pmbrull I left one comment. 
[nit] Could we fix the type annotation code smell? 🙂

File: ingestion/setup.py [hunk 2]
```diff
@@ -180,7 +180,7 @@ def get_long_description():
     "kafka": {*COMMONS["kafka"]},
     "kinesis": {VERSIONS["boto3"]},
     "ldap-users": {"ldap3==2.9.1"},
-    "looker": {"looker-sdk>=22.20.0"},
+    "looker": {"looker-sdk>=22.20.0", "lkml~=1.3"},
```
[the remark this answers is not in the document]
Comment (pmbrull) [reply ingestion/setup.py:183]: yup anything less than 2 but more than 1.3

**Comment by pmbrull** [comment]
> @pmbrull I left one comment. [nit] Could we fix the type annotation code smell? 🙂

Hi @TeddyCr, actually I rather stick to `Iterable` rather than going the `Generator` route 😅 

This is how we are typing all the functions in the services ([example](https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/source/database/database_service.py#L284))

Typing a Generator is actually quite verbose and a PITA ([docs](https://docs.python.org/3.6/library/typing.html#typing.Generator)), as we would need something like `Generator[YieldType, SendType, ReturnType]`. In our case, we are just interested on the `YieldType`, so we can bypass the type check by simplifying this with `Iterator[YieldType]`. This is a documented alternative highlighted in the docs above.

WDYT?

**Pull request merged**